src/components/SearchBar.jsx

```diff
@@ -7,13 +7,68 @@ const INTENT_VERB = 'OPEN'
 const INTENT_DOCTYPE = 'io.cozy.suggestions'
 const SUGGESTIONS_PER_SOURCE = 10
 
+const normalizeString = str => str.toString().toLowerCase().replace(/\//g, ' ').normalize('NFD').replace(/[\u0300-\u036f]/g, '').split(' ')
+
+const highlightQueryTerms = (searchResult, query) => {
+  const normalizedQueryTerms = normalizeString(query)
+  const normalizedResultTerms = normalizeString(searchResult)
+
+  const matchedIntervals = []
+  const spacerLength = 1
+  let currentIndex = 0
+
+  normalizedResultTerms.forEach(resultTerm => {
+    normalizedQueryTerms.forEach(queryTerm => {
+      const index = resultTerm.indexOf(queryTerm)
+      if (index >= 0) {
+        matchedIntervals.push({
```

Comment:
don'tlike `forEach` with side effects. Can't we have a `map`?

Reply:
No, precisely because of the "side effects", since they aren't side effects at all. They aren't a byproduct of the `forEach`, they are the reason for it.
More extensively, I'm creating new information based on 2 existing arrays, not reducing or transforming the existing data. The array I'm building could have less, the same, or more elements than any of the base arrays. And the data it contains is not a transformation of the values of the existing arrays either.

Reply:
When you say:
> No, precisely because of the "side effects", since they aren't side effects at all. 

You mean the code could be refactored like this:
```js
normalizedResultTerms.forEach(newFunction1())

function newFunction1() {
  return resultTerm => {
    normalizedQueryTerms.forEach(newFunction(resultTerm));
    currentIndex += resultTerm.length + spacerLength;
  };
}

function newFunction(resultTerm) {
  return queryTerm => {
    const index = resultTerm.indexOf(queryTerm)
    if (index >= 0) {
      matchedIntervals.push({
        from: currentIndex + index,
        to: currentIndex + index + queryTerm.length
      })
    }
  }
}
```
with `newFunction` and newFunction1` as [pure functions](https://en.wikipedia.org/wiki/Pure_function) ?

I mean: you need to be opinionated to write some code. But the reason you give to justify the point is not good, nor real.
Nevertheless you can keep this code instead of writing a real pure functional version. There is no problem to have this code on the codebase. It is just a matter of readability and maintainability.